controller/selector.go

```diff
@@ -0,0 +1,22 @@
+package controller
+
+import (
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type Selector interface {
+	Matches(labels Labels) bool
+}
```

Comment:
This new implementation is just leaned out implementation of the [apiextensions Selector type](https://pkg.go.dev/k8s.io/apimachinery@v0.16.6/pkg/labels?tab=doc#Selector).

Now with the leaner interface we can confidently implement more robust conditions. Including OR conditional for labels existence.